scripts/cybershake/queue_monitor.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+"""Script for continuously updating the slurm mgmt db from the queue."""
+import os
+import json
+import argparse
+import time
+from typing import List
+
+import qcore.simulation_structure as sim_struct
+from scripts.management.MgmtDB import MgmtDB, SlurmTask
+
+
+# Have to include sub-seconds, as clean up can run sub one second.
+DATE_FORMAT = "%Y%m%d%H%M%S_%f"
+
+
+def get_queue_entries(entry_files: List[str]):
+    queue_entries = []
+    for entry_file in entry_files:
+        try:
+            with open(entry_file, "r") as f:
+                data_dict = json.load(f)
+        except json.JSONDecodeError as ex:
+            print(
+                "Failed to decode the file {} as json. Check that this is "
+                "valid json. Ignored!".format(
+                    entry_file
+                )
+            )
+            continue
+
+        queue_entries.append(
+            SlurmTask(
+                run_name=os.path.basename(entry_file).split(".")[1],
+                proc_type=data_dict[MgmtDB.col_proc_type],
+                status=data_dict[MgmtDB.col_status],
+                job_id=data_dict[MgmtDB.col_job_id],
+                retries=data_dict[MgmtDB.col_retries],
+                error=data_dict.get("error")
+            )
+        )
+
+    return queue_entries
+
+
+def main(args):
+    mgmt_db = MgmtDB(sim_struct.get_mgmt_db(args.root_folder))
+    queue_folder = sim_struct.get_mgmt_db_queue(args.root_folder)
+
```

Comment:
Probably should print instructions on how to close file. Or have an end condition for the while.

Reply:
Add a signal handler to ensure graceful close on ctrl+c?